opencog/atoms/distvalue/DistributionalValue.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * opencog/truthvalue/DistributionalValue.h
+ *
+ * Copyright (C) 2018 SingularityNet
+ * All Rights Reserved
+ *
+ * Written by Roman Treutlein
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _OPENCOG_DISTRIBUTIONAL_VALUE_H
+#define _OPENCOG_DISTRIBUTIONAL_VALUE_H
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <limits>
+
+#include <opencog/util/Counter.h>
+#include <opencog/atoms/distvalue/ConditionalDV.h>
+
+/** \addtogroup grp_atomspace
+ *	@{
+ */
+
+namespace opencog
+{
+
+typedef double strength_t;
+typedef double confidence_t;
+typedef double count_t;
+
+class DistributionalValue;
+
+typedef std::vector<double> DVec;
+
+typedef std::vector<double> Interval;
+
+typedef std::vector<Interval> DVKey;
+
+typedef std::vector<DVKey> DVKeySeq;
+
+typedef Counter<DVKey, double> DVCounter;
+
+typedef std::shared_ptr<const DistributionalValue> DistributionalValuePtr;
+
+class AtomSpace;
+
+class DistributionalValue
+	: public Value
+{
+
+	friend class ConditionalDV;
+
+	DVCounter _value;
+
+	// Disallow assignment -- truth values are immutable!
+	DistributionalValue& operator=(const DistributionalValue& rhs) {
+		throw RuntimeException(TRACE_INFO, "Cannot modify truth values!");
+	}
+
+
+public:
+	static count_t DEFAULT_K;
+
+	DistributionalValue();
+	DistributionalValue(DVCounter);
```

Comment:
```suggestion
	DistributionalValue(const DVCounter&);
```
This will avoid to copy twice.